### DISCUSSION TITLE
"Hostname or IP for server" configuration question

### DISCUSSION THREAD

**Discussion by chrisjdixon** [dokku/dokku#4827] [Q&A]
I'm trying to install Dokku and am trying to meticulously follow instructions to the letter because my previous installation failed after unsuccessfully trying to [diagnose why I only saw the default nginx page](https://dokku.com/docs/getting-started/troubleshooting/#i-deployed-my-app-but-i-am-getting-the-default-nginx-page). I'm running through [these instructions](https://dokku.com/#install-apt) and have encountered a super basic question I'm unsure how to answer: [Hosname or IP for server:](https://imgur.com/VdeFuY6).

Is this asking what should be the default external facing url that users will ultimately visit or is this some internal thing? I know this is probably super obvious but I'm new to this entire world so I'm potato and previously failed installation has left me wary of making assumptions.

Also, I was previously working off the instructions on the [Getting Started with Dokku page](https://dokku.com/docs/getting-started/installation/#installing-the-latest-stable-version), but someone recommended I work off the instructions from the earlier linked simpler page. Why do these instructions differ? I definitely didn't see this purple screen when working on it previously. Which is the more appropriate to use for a beginner?

Apologies for these super stupid questions.

**Comment by josegonzalez** [comment]
The bootstrap script wraps the apt calls and sets some defaults, so basically skips the dpkg instructions. The apt instructions don't do so and therefore ask for more information. I've documented both in case folks want to use the apt instructions for server automation.

Regarding the question, it should quite literally be the ip address of the server or the hostname you are going to add to the server. If you don't have a hostname now, you can just use the ip address. The value can be changed later via `dokku domains:set-global` (or `domains:add-global` I guess). The value here is used for app subdomains, so if you deploy an app named `test-app` and set a hostname `dokku.me`, the app domain gets set to `test-app.dokku.me`.
